react-carousel/my-app/src/Carousel.tsx

```diff
@@ -22,7 +22,7 @@ export function Carousel({ images }: Props) {
       setCurrentImage((currentImage + 1) % images.length);
     }, 3000);
     return () => clearTimeout(timerId);
-  });
+  }, [currentImage, setCurrentImage, images.length]);
```

Comment:
`setCurrentImage` isn't needed here since it is a state setter. But well done on identifying them all!